webapp/components/test-runs-query.html

```diff
@@ -143,30 +143,28 @@
          * Update this builder's properties to match the given query params.
          */
         updateQueryParams(params) {
-          // Expand a global channel label into the separate products
-          let sharedChannel = (params.label || []).find(l => CHANNELS.has(l));
-          if (sharedChannel) {
-            params.label = params.label.filter(l => !CHANNELS.has(l));
-          }
-
           this.clearQuery();
           if (!params) {
             return;
           }
           if (!this.computeIsLatest(params.sha)) {
             this.sha = params.sha;
           }
-          if ('label' in params) {
-            this.labels = params.label;
-          }
           if ('product' in params) {
             this.products = params.product.map(p => this.parseProductSpec(p));
           }
+          // Expand a global channel label into the separate products
+          let sharedChannel = (params.label || []).find(l => CHANNELS.has(l));
```

Comment:
Nit: if you just `let sharedChannel` here you can move `params.label.find(l => CHANNELS.has(l))` into the if body. Minimally clearer, take your pick :)